desc/objectives/utils.py

```diff
@@ -81,19 +82,43 @@ def factorize_linear_constraints(constraints, objective):  # noqa: C901
         xz = _tree_zeros_like([t.params_dict for t in con.things])
         # computing A matrix for each constraint for each thing in the optimization
         for thing in objective.things:
-            if thing in con.things:
-                A_ = con.jac_scaled(*xz)[con.things.index(thing)]
+            if not isinstance(thing, OptimizableCollection):
+                if thing in con.things:
+                    A_ = con.jac_scaled(*xz)[con.things.index(thing)]
+                else:
+                    A_ = {
+                        arg: jnp.zeros((con.dim_f, dimx))
+                        for arg, dimx in thing.dimensions.items()
+                    }
+                args = (
+                    objective._args
+                    if prox_flag and isinstance(thing, Equilibrium)
+                    else thing.optimizable_params
+                )
+                A_per_thing.append(jnp.hstack([A_[arg] for arg in args]))
+
             else:
-                A_ = {
-                    arg: jnp.zeros((con.dim_f, dimx))
-                    for arg, dimx in thing.dimensions.items()
-                }
-            args = (
-                objective._args
-                if prox_flag and isinstance(thing, Equilibrium)
-                else thing.optimizable_params
-            )
-            A_per_thing.append(jnp.hstack([A_[arg] for arg in args]))
+                A_per_subthing = []
+                # it is an optimizable collection, loop through its sub-objects
+                for subthing in thing:
+                    if subthing in con._subthings:
+                        # this returns a tuple of length 1 for some reason?
+                        A_ = con.jac_scaled(*xz)[0][con._subthings.index(subthing)]
+                    else:
+                        A_ = {
```

Comment:
This is completely untested right now (way I have it setup as of commit 237da24f3a is that `FixParameter` has in its `self._subthings` list every single object contained in the `OptimizableCollection` it has as its `self.things[0]`, so even though I am not fixing anything for the VF field in the example, it is still counted in the first logic branch above this line and the else is not triggered